R/z_pages.R

```diff
@@ -141,3 +141,83 @@ check_no_github_repo <- function(owner, repo) {
     "http_error_404" = function(err) FALSE
   )
 }
+
+get_pages_info <- function(viz_user_repo){
+  viz_dir <- tempfile()
+  origin_url <- paste0("https://github.com/", viz_user_repo, ".git")
+  gert::git_clone(origin_url, viz_dir)
+  gert::git_branch_checkout("gh-pages", repo=viz_dir)
+  Capture.PNG <- file.path(viz_dir, "Capture.PNG")
+  if(!file.exists(Capture.PNG)){
+    stop(sprintf("gh-pages branch of %s should contain file named Capture.PNG (screenshot of data viz)", viz_user_repo))
+  }
+  plot.json <- file.path(viz_dir, "plot.json")
+  jlist <- RJSONIO::fromJSON(plot.json)
+  commit.row <- gert::git_log(max=1, repo=viz_dir)
+  repo.row <- data.table(
+    viz_user_repo, Capture.PNG, commit.POSIXct=commit.row$time)
+  to.check <- c(
+    source="URL of data viz source code",
+    title="string describing the data viz")
+  for(attr.name in names(to.check)){
+    attr.value <- jlist[[attr.name]]
+    if(
+      is.character(attr.value)
+      && length(attr.value)==1
+      && !is.na(attr.value)
+      && nchar(attr.value)>0
+    ){
+      set(repo.row, j=attr.name, value=attr.value)
+    }else{
+      stop(sprintf("plot.json file in gh-pages branch of %s should have element named %s which should be %s", viz_user_repo, attr.name, to.check[[attr.name]]))
+    }
+  }
+  repo.row
+}
+
+##' A gallery is a collection of animints that have been published to
+##' github pages. First repos.txt is read, then we clone each repo
+##' which is not already present in meta.csv, and parse meta-data
+##' (title, source, Capture.PNG) from the gh-pages branch, and
+##' write/commit the data, re-render index.Rmd in gallery, and push
+##' gallery to origin.
+##' @title Update gallery
+##' @param gallery_path path to local github repo with gh-pages active.
+##' @return named list of data tables (meta and error).
+##' @author Toby Dylan Hocking
+##' @export
+update_gallery <- function(gallery_path="~/R/gallery"){
```

Comment:
here is a new function for updating the gallery at https://github.com/animint/gallery/tree/gh-pages
please code review @Faye-yufan 
probably do not want to test this, because it actually pushes to github?

Reply:
actually I could add a test to make sure that it updates the meta.csv / error.csv files